src/forms/FormFieldAdvancedSelection/AdvancedSelectionDrawer.tsx

```diff
@@ -348,8 +350,9 @@ const AdvancedSelectionDrawer = (props: AdvanceSelectionDrawerPropTypes): ReactE
 					size: "100%",
 					onChangeCb: checkboxListChanged,
 					inputSettings: {
-						options: fieldDef?.inputSettings?.options && filteredList,
-						getOptions: fieldDef?.inputSettings?.getOptions && (() => filteredList),
+						options:
+							(fieldDef?.inputSettings?.options && filteredList) ||
+							(fieldDef?.inputSettings?.getOptions && (() => filteredList)),

```

Comment:
I'm not sure why I can't comment on that line, but I believe we can delete the getMoreOptions from the dependency array, it's not being used here and when doing the research / testing if I remember correctly it caused re-renders on the fields.

Reply:
If we removed ``getMoreOptions`` as a dependency we lost the ``selectLimit`` functionality. I tried to put ``fieldDef?.inputSettings?.selectLimit`` as a dependency but it did not work